deep_agent/aegra/auth.py

```
@@ -171,7 +172,43 @@ async def authenticate(headers: dict) -> dict:
+    # This ensures only authorized callers can access this specific agent.
     auth_header = headers.get("authorization", "")
+
+    if INTERNAL_API_SECRET:
+        request_secret = headers.get("x-internal-secret", "") or headers.get(
+            "x-gitlab-token", ""
+        )
+        if not request_secret:
+            # No secret provided but agent requires one — only allow if Bearer token present
+            # (direct SSO access without agent-level secret)
+            if not auth_header.startswith("Bearer "):
+                raise PermissionError("Missing or invalid Authorization header")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)